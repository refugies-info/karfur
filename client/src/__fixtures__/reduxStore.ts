import { RootState } from "services/rootReducer";
import { SearchResultsState } from "services/SearchResults/searchResults.reducer";
import { UserFavoritesState } from "services/UserFavoritesInLocale/UserFavoritesInLocale.reducer";
import { activeThemesMock } from "./activeThemes";

const initialMockUserState = {
  user: null,
  admin: false,
  traducteur: false,
  expertTrad: false,
  contributeur: false,
  membreStruct: false,
  userId: "",
  userFetched: false,
  rolesInStructure: [],
};
const initialMockLangueState = {
  langues: [],
  languei18nCode: "fr",
  showLanguageModal: false,
  showLangModal: false,
};

const initialMockTranslationState = {
  translation: {
    initialText: {},
    translatedText: {},
  },
  translations: [],
};

const initialUserFavoritesState: UserFavoritesState = {
  favorites: [],
  showFavoriteModal: false
}

const initialMockSearchReults: SearchResultsState = {
  results: {
    dispositifs: [],
    demarches: [],
    dispositifsSecondaryTheme: []
  },
  query: {
    search: "",
    departments: [],
    themes: [],
    needs: [],
    age: [],
    frenchLevel: [],
    language: [],
    sort: "date",
    type: "all",
  }
}

export const initialMockStore = {
  user: initialMockUserState,
  langue: initialMockLangueState,
  activeDispositifs: [],
  tts: { ttsActive: false, showAudioSpinner: false },
  userStructure: null,
  selectedDispositif: null,
  loadingStatus: {},
  translation: initialMockTranslationState,
  activeStructures: [],
  selectedStructure: null,
  allDispositifs: [],
  allStructures: [],
  users: [],
  userFavorites: initialUserFavoritesState,
  userContributions: [],
  needs: [],
  themes: {
    activeThemes: activeThemesMock,
    inactiveThemes: []
  },
  searchResults: initialMockSearchReults
};
