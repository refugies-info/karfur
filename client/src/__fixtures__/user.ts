import { GetUserInfoResponse, UserStatus } from "api-types";
import moment from "moment";

const date = moment("2019-05-01 12:00:00");
export const testUser: GetUserInfoResponse = {
  username: "test",
  _id: "testObjectId",
  structures: [],
  contributions: [],
  phone: "",
  roles: [],
  email: "",
  selectedLanguages: [],
  status: UserStatus.ACTIVE,
  traductionsFaites: []
};

export const testUserWithRoles: GetUserInfoResponse = {
  ...testUser,
  structures: ["testObjectId"],
  roles: [
    { nom: "Admin", _id: "testObjectId", nomPublic: "Admin" },
    { nom: "Trad", _id: "testObjectId", nomPublic: "Trad" },
    {
      nom: "ExpertTrad",
      _id: "testObjectId",
      nomPublic: "ExpertTrad"
    },
    {
      nom: "Contrib",
      _id: "testObjectId",
      nomPublic: "Contrib"
    }
  ]
};
