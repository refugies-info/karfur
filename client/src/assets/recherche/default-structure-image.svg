<svg fill="none" height="42" viewBox="0 0 42 42" width="42" xmlns="http://www.w3.org/2000/svg"><rect fill="#eee7e5" height="41.6667" rx="8" width="41.6667" x=".166992" y=".162109"/><path d="m22.0729 10.4441-11.4479-1.73954c-.5313-.08333-1.07292.0625-1.47917.40625-.40625.35417-.64583.86458-.64583 1.39579v21.948c0 .5729.46875 1.0416 1.04167 1.0416h3.38543v-5.4687c0-1.0104.8125-1.8229 1.8229-1.8229h2.6042c1.0104 0 1.8229.8125 1.8229 1.8229v5.4687h4.4271v-21.25c0-.8958-.6459-1.6562-1.5313-1.8021zm-7.8437 13.4167h-1.5625c-.4313 0-.7813-.35-.7813-.7812 0-.4313.35-.7813.7813-.7813h1.5625c.4312 0 .7812.35.7812.7813 0 .4312-.35.7812-.7812.7812zm0-3.125h-1.5625c-.4313 0-.7813-.35-.7813-.7812 0-.4313.35-.7813.7813-.7813h1.5625c.4312 0 .7812.35.7812.7813 0 .4312-.35.7812-.7812.7812zm0-3.125h-1.5625c-.4313 0-.7813-.35-.7813-.7812 0-.4313.35-.7813.7813-.7813h1.5625c.4312 0 .7812.35.7812.7813 0 .4312-.35.7812-.7812.7812zm0-3.125h-1.5625c-.4313 0-.7813-.35-.7813-.7812 0-.4313.35-.7813.7813-.7813h1.5625c.4312 0 .7812.35.7812.7813 0 .4312-.35.7812-.7812.7812zm5.2083 9.375h-1.5625c-.4313 0-.7812-.35-.7812-.7812 0-.4313.3499-.7813.7812-.7813h1.5625c.4312 0 .7813.35.7813.7813 0 .4312-.3501.7812-.7813.7812zm0-3.125h-1.5625c-.4313 0-.7812-.35-.7812-.7812 0-.4313.3499-.7813.7812-.7813h1.5625c.4312 0 .7813.35.7813.7813 0 .4312-.3501.7812-.7813.7812zm0-3.125h-1.5625c-.4313 0-.7812-.35-.7812-.7812 0-.4313.3499-.7813.7812-.7813h1.5625c.4312 0 .7813.35.7813.7813 0 .4312-.3501.7812-.7813.7812zm0-3.125h-1.5625c-.4313 0-.7812-.35-.7812-.7812 0-.4313.3499-.7813.7812-.7813h1.5625c.4312 0 .7813.35.7813.7813 0 .4312-.3501.7812-.7813.7812z" fill="#5a483c"/><path d="m32.0632 19.7895-7.4167-1.5532v15.2594h7.0312c1.0052 0 1.823-.8177 1.823-1.8229v-10.1021c0-.8604-.5865-1.5927-1.4375-1.7812zm-2.4688 10.5812h-1.5625c-.4312 0-.7812-.35-.7812-.7812 0-.4313.35-.7813.7812-.7813h1.5625c.4313 0 .7813.35.7813.7813 0 .4312-.35.7812-.7813.7812zm0-3.125h-1.5625c-.4312 0-.7812-.35-.7812-.7812 0-.4313.35-.7813.7812-.7813h1.5625c.4313 0 .7813.35.7813.7813 0 .4312-.35.7812-.7813.7812zm0-3.125h-1.5625c-.4312 0-.7812-.35-.7812-.7812 0-.4313.35-.7813.7812-.7813h1.5625c.4313 0 .7813.35.7813.7813 0 .4312-.35.7812-.7813.7812z" fill="#fca306"/></svg>