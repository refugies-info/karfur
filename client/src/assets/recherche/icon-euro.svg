<svg fill="none" height="16" viewBox="0 0 16 16" width="16" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><clipPath id="a"><path d="m0 0h16v16h-16z"/></clipPath><g clip-path="url(#a)"><path d="m8.00065 15.6673c1.51632 0 2.99855-.4496 4.25935-1.292 1.2608-.8425 2.2435-2.0398 2.8237-3.4407.5803-1.40094.7321-2.94245.4363-4.42964s-1.026-2.85325-2.0982-3.92546c-1.0722-1.0722-2.4383-1.802379-3.92546-2.098199-1.48718-.29582-3.02869-.143995-4.4296.436277-1.4009.580272-2.59827 1.562932-3.44069 2.823702-.842424 1.26078-1.292066 2.74305-1.292066 4.25937 0 2.03335.807736 3.98335 2.245516 5.42115s3.38782 2.2455 5.42115 2.2455z" fill="#ffeb3b"/><path d="m8.00002.33399c-1.52772.000127-3.02056.456669-4.28703 1.31106-1.26646.85439-2.24879 2.06766-2.820972 3.48418-.57218 1.41652-.708118 2.97168-.390374 4.46599.317744 1.49428 1.074676 2.85958 2.173706 3.92078l10.83995-10.84001c-.7141-.74174-1.5708-1.33154-2.5186-1.733994-.9477-.402452-1.96702-.609258-2.99668-.608006z" fill="#f9ef99"/><g stroke-linecap="round" stroke-linejoin="round"><path d="m8.00065 15.6673c1.51632 0 2.99855-.4496 4.25935-1.292 1.2608-.8425 2.2435-2.0398 2.8237-3.4407.5803-1.40094.7321-2.94245.4363-4.42964s-1.026-2.85326-2.0982-3.92546-2.4383-1.802379-3.92546-2.098199c-1.48718-.29582-3.02869-.143995-4.4296.436277-1.4009.580272-2.59827 1.562932-3.44069 2.823702-.842424 1.26078-1.292066 2.74305-1.292066 4.25937 0 2.03335.807736 3.98335 2.245516 5.42115s3.38782 2.2455 5.42115 2.2455z" stroke="#f9aa38" stroke-width=".666667"/><path d="m9.66732 11.6673c-2.39334 0-4.33334-.6666-4.33334-3.66665 0-3 1.94-3.66667 4.33334-3.66667m-5.33334 4.66667h3.66667m-3.66667-1.33333h4.33334" stroke="#443023"/></g></g></svg>