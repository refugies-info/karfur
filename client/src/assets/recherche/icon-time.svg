<svg fill="none" height="16" viewBox="0 0 16 16" width="16" xmlns="http://www.w3.org/2000/svg"><path d="m8.66602 13c2.76138 0 4.99998-2.2386 4.99998-5 0-2.76142-2.2386-5-4.99998-5-2.76143 0-5 2.23858-5 5 0 2.7614 2.23857 5 5 5z" fill="#d2edfc"/><g stroke="#443023" stroke-linecap="round" stroke-linejoin="round" stroke-width=".666667"><path d="m8.66732 4.66602v3.66666h3.33338m-11.666716-2.33733 1.999996 3 2.33334-2.66667"/><path d="m8.66732 14.3327c1.25261 0 2.47708-.3715 3.51858-1.0674s1.8533-1.685 2.3327-2.8423c.4793-1.15725.6047-2.43067.3604-3.65922-.2444-1.22855-.8476-2.35704-1.7333-3.24277-.8858-.88573-2.0143-1.48893-3.24281-1.7333-1.22855-.24437-2.50197-.11895-3.65923.3604-1.15727.47936-2.1464 1.29112-2.84232 2.33263-.69591 1.04151-1.06736 2.266-1.06736 3.51861v.996"/></g></svg>