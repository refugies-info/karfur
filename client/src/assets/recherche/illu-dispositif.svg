<svg height="160" viewBox="0 0 160 160" width="160" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd" transform="translate(0 2)"><path d="m37.0126331 103.032994c15.7122992-6.3885731 36.867451-5.1061134 45.4810145-3.666449 20.5226444 1.08558 50.3078074 11.864621 62.4485594 19.754764 12.139754 7.890142 18.759898 20.614056 12.885405 26.986979-4.522751 4.906542-11.90948 7.769522-27.762536 9.144191-23.057359 1.998704-54.4950161-4.577578-66.639356-5.165725s-55.18514212.620046-60.55751674-4.539698c-5.37237462-5.159745-2.85360986-12.267353 4.69082181-19.116774 7.54433203-6.850418 9.81328373-15.41045 29.45360803-23.397288z" fill="#feebde" fill-rule="nonzero"/><path d="m71.382054 130.912162v-6.383888c-1.1291427-7.775503-3.237002-20.033884-6.3386304-29.0993238-3.9396883-11.5147235-15.9934137-31.2718796-20.5617214-38.5661993-4.191794-5.8854581-6.6572268-13.0857739-6.6572268-20.8618757 0-19.8827605 16.1181207-36.0008752 36.0008811-36.0008752 19.8832589 0 36.0004825 16.1181147 36.0004825 36.0008752 0 7.7838773-2.470218 14.9907723-6.66899 20.8795202-4.7151449 7.4612936-16.8007698 27.1175675-20.7118482 38.5485548-3.1016285 9.0654398-5.2094878 21.3238208-6.3386305 29.0993238v6.383888z" fill="#dd0539"/><path d="m73.8318361 52.3656189c7.2300222 0 13.091157-5.8611348 13.091157-13.0912567s-5.8611348-13.0912567-13.091157-13.0912567c-7.2301219 0-13.0912567 5.8611348-13.0912567 13.0912567s5.8611348 13.0912567 13.0912567 13.0912567z" fill="#feebde"/></g></svg>