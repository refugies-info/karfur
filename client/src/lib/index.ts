export const jsUcfirst = (string: string) =>
  string &&
  string.length > 1 &&
  string.charAt(0).toUpperCase() + string.slice(1, string.length);

export const jsLcfirst = (string: string) =>
  string &&
  string.length > 1 &&
  string.charAt(0).toLowerCase() + string.slice(1, string.length);

export const limitNbCaracters = (string: string, nbCaractersMax: number) =>
  string.substring(0, Math.min(string.length, nbCaractersMax)) +
  (string.length > nbCaractersMax ? "..." : "");

export const removeAccents = (str = "") => {
  var accents =
    "ÀÁÂÃÄÅàáâãäåÒÓÔÕÕÖØòóôõöøÈÉÊËèéêëðÇçÐÌÍÎÏìíîïÙÚÛÜùúûüÑñŠšŸÿýŽž";
  var accentsOut =
    "AAAAAAaaaaaaOOOOOOOooooooEEEEeeeeeCcDIIIIiiiiUUUUuuuuNnSsYyyZz";
  const chars = str.split("");
  let i, x;
  for (i = 0; i < chars.length; i++) {
    if ((x = accents.indexOf(chars[i])) !== -1) {
      chars[i] = accentsOut[x];
    }
  }
  return chars.join("");
};

// rules for password check :
// nbNumUpper = nbNumeric + nbUpper
// score 0 (cannot set password): length < 5
// score 0.5 : 7>length>=5 + nbNumUpper = 1 or 2 or0
// score 1.5 : 7>length >= 5 + nbNumUpper>=3 OR length>=7 + nbNumUpper<3
// score 2.5 : length >= 7 + nbNumUpper>=3
// score 3.5 : length >= 7 + nbNumUpper>=3 + at least 1 special caractere

// make sure to use same function in backend
export const computePasswordStrengthScore = (
  password: string
): { score: number } => {
  if (password.length < 5) {
    return { score: 0 };
  }

  const numerics = password.match(/\d+/g)?.join("");
  const nbNumeric = numerics ? numerics.length : 0;

  const upperCase = password.match(/[A-Z]/g)?.join("");
  const nbUpperCase = upperCase ? upperCase.length : 0;

  if (password.length < 7) {
    if (nbUpperCase + nbNumeric < 3) {
      return { score: 1 };
    }
    return { score: 2 };
  }

  if (nbNumeric + nbUpperCase < 3) {
    return { score: 2 };
  }
  const specialCar = password.match(/[!@#$&*]/g)?.join("");
  const nbSpecialCar = specialCar ? specialCar.length : 0;
  if (nbSpecialCar > 0) {
    return { score: 4 };
  }

  return { score: 3 };
};
