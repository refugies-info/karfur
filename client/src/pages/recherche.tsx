import React, { useEffect } from "react";
import { END } from "redux-saga";
import SEO from "components/Seo";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
import { wrapper } from "services/configureStore";
import { toggleLangueActionCreator } from "services/Langue/langue.actions";
import { fetchActiveDispositifsActionsCreator } from "services/ActiveDispositifs/activeDispositifs.actions";
import { fetchThemesActionCreator } from "services/Themes/themes.actions";
import { cls } from "lib/classname";
import { getLanguageFromLocale } from "lib/getLanguageFromLocale";
import styles from "scss/pages/recherche.module.scss";
import SearchHeader from "components/Pages/recherche/SearchHeader";
import { useSelector } from "react-redux";
import DemarcheCard from "components/Pages/recherche/DemarcheCard";
import { activeDispositifsSelector } from "services/ActiveDispositifs/activeDispositifs.selector";
import { Container } from "reactstrap";
import DispositifCard from "components/Pages/recherche/DispositifCard";

export type SearchQuery = {
  departments?: string[];
  needs?: string[];
  theme?: string;
  keyword?: string;
  age?: string;
  frenchLevel?: string;
  langue?: string;
  order: "date" | "views" | "theme" | "";
  type?: "dispositifs" | "demarches";
};

const Recherche = () => {
  const dispositifs = useSelector(activeDispositifsSelector);

  return (
    <div className={cls(styles.container)}>
      <SEO title="Recherche" />
      <SearchHeader />

      <Container>
        <div className="d-flex flex-wrap">
          {dispositifs
            .filter((d) => d.typeContenu === "dispositif")
            .slice(0, 10)
            .map((d, i) => (
              <DispositifCard key={i} dispositif={d} />
            ))}
        </div>

        <div className="d-flex flex-wrap">
          {dispositifs
            .filter((d) => d.typeContenu === "demarche")
            .slice(0, 10)
            .map((d, i) => (
              <DemarcheCard key={i} demarche={d} />
            ))}
        </div>
      </Container>
    </div>
  );
};

export const getServerSideProps = wrapper.getServerSideProps((store) => async ({ locale }) => {
  if (locale) {
    store.dispatch(toggleLangueActionCreator(locale)); // will fetch dispositifs automatically
  } else {
    store.dispatch(fetchActiveDispositifsActionsCreator());
  }
  store.dispatch(fetchThemesActionCreator());
  store.dispatch(END);
  await store.sagaTask?.toPromise();

  return {
    props: {
      ...(await serverSideTranslations(getLanguageFromLocale(locale), ["common"]))
    }
  };
});

export default Recherche;
