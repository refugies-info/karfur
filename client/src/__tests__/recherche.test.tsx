import { wrapWithProvidersAndRender } from "../../jest/lib/wrapWithProvidersAndRender";
import recherche from "../pages/recherche";
import { initialMockStore } from "__fixtures__/reduxStore";
import { act, ReactTestRenderer } from "react-test-renderer";
import { setupGoogleMock } from "../__mocks__/react-google-autocomplete";
import "jest-styled-components";

jest.mock("next/router", () => require("next-router-mock"));

describe("recherche", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    setupGoogleMock();
  });

  let component: ReactTestRenderer;

  it("renders search results", () => {
    window.scrollTo = jest.fn();

    act(() => {
      component = wrapWithProvidersAndRender({
        Component: recherche,
        reduxState: {
          ...initialMockStore,
          searchResults: {
            results: {
              dispositifs: [],
              demarches: [],
              dispositifsSecondaryTheme: [],
            },
            query: {
              search: "",
              departments: [],
              themes: ["6319f6b363ab2bbb162d7df5"],
              needs: [],
              age: [],
              frenchLevel: [],
              language: [],
              sort: "date",
              type: "all",
            },
            inputFocused: {
              search: false,
              location: false,
              theme: false,
            },
          },
        },
      });
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
