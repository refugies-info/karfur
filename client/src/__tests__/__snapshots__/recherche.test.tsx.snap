// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recherche renders search results 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="title"
    >
      <div
        class="container"
      >
        <h1>
          Recherche.title
        </h1>
        <p>
          Recherche.subtitle
        </p>
      </div>
    </div>
    <div
      class="filters"
    >
      <div
        class="stickybar"
      >
        <div
          class="container container"
        >
          <div
            class="container"
          >
            <div
              class="zone"
            >
              <i
                class="fr-icon-search-line icon"
              />
              <input
                class="input"
                placeholder="Mot-clé"
                type="text"
                value=""
              />
            </div>
            <div
              class="border"
            >
               
            </div>
          </div>
          <div
            class="d-flex align-items-center gap-3"
          >
            <span
              class="container"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="button"
                data-state="closed"
                id="radix-:r0:"
                type="button"
              >
                Dispositif.Département
                <i
                  class="ri-arrow-down-s-line"
                />
              </button>
            </span>
            <span
              class="container"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="button values"
                data-state="closed"
                id="radix-:r2:"
                type="button"
              >
                Français
              </button>
              <button
                class="clear"
                title="Recherche.resetButton"
              >
                <i
                  class="ri-close-circle-fill"
                />
              </button>
            </span>
            <span
              class="container"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="button"
                data-state="closed"
                id="radix-:r4:"
                type="button"
              >
                Recherche.filterStatus
                <i
                  class="ri-arrow-down-s-line"
                />
              </button>
            </span>
            <span
              class="container"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="button"
                data-state="closed"
                id="radix-:r6:"
                type="button"
              >
                Recherche.filterPublic
                <i
                  class="ri-arrow-down-s-line"
                />
              </button>
            </span>
            <span
              class="container"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="button"
                data-state="closed"
                id="radix-:r8:"
                type="button"
              >
                Recherche.filterAge
                <i
                  class="ri-arrow-down-s-line"
                />
              </button>
            </span>
            <span
              class="container"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="button"
                data-state="closed"
                id="radix-:ra:"
                type="button"
              >
                Recherche.filterFrenchLevel
                <i
                  class="ri-arrow-down-s-line"
                />
              </button>
            </span>
            <span
              class="container"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="button"
                data-state="closed"
                id="radix-:rc:"
                type="button"
              >
                Recherche.filterLanguage
                <i
                  class="ri-arrow-down-s-line"
                />
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container no_result"
    >
      <div
        class="container_inner container"
      >
        <div
          class="types"
        >
          <button
            class="btn selected btn btn-secondary"
            type="button"
          >
            Filters.typeAll (0)
          </button>
          <button
            class="btn btn btn-secondary"
            type="button"
          >
            Filters.typeDispositif (0)
          </button>
          <button
            class="btn btn btn-secondary"
            type="button"
          >
            Filters.typeDemarche (0)
          </button>
        </div>
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown btn btn-secondary"
            type="button"
          >
            <span
              class="icon icon"
            >
              <i
                style="height: 20px; line-height: 20px;"
              >
                <svg
                  class="eva eva-swap-outline"
                  fill="black"
                  height="20px"
                  viewBox="0 0 24 24"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    data-name="Layer 2"
                  >
                    <g
                      data-name="swap"
                    >
                      <rect
                        height="24"
                        opacity="0"
                        transform="rotate(-90 12 12)"
                        width="24"
                      />
                      <path
                        d="M4 9h13l-1.6 1.2a1 1 0 0 0-.2 1.4 1 1 0 0 0 .8.4 1 1 0 0 0 .6-.2l4-3a1 1 0 0 0 0-1.59l-3.86-3a1 1 0 0 0-1.23 1.58L17.08 7H4a1 1 0 0 0 0 2z"
                      />
                      <path
                        d="M20 16H7l1.6-1.2a1 1 0 0 0-1.2-1.6l-4 3a1 1 0 0 0 0 1.59l3.86 3a1 1 0 0 0 .61.21 1 1 0 0 0 .79-.39 1 1 0 0 0-.17-1.4L6.92 18H20a1 1 0 0 0 0-2z"
                      />
                    </g>
                  </g>
                </svg>
              </i>
            </span>
            Filters.sortDate
          </button>
          <div
            aria-hidden="true"
            class="menu dropdown-menu"
            data-bs-popper="static"
            role="menu"
            tabindex="-1"
          >
            <button
              class="item selected dropdown-item"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              Filters.sortDate
              <span
                class="icon"
              >
                <i
                  style="height: 20px; line-height: 20px;"
                >
                  <svg
                    class="eva eva-checkmark-outline"
                    fill="white"
                    height="20px"
                    viewBox="0 0 24 24"
                    width="20px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      data-name="Layer 2"
                    >
                      <g
                        data-name="checkmark"
                      >
                        <rect
                          height="24"
                          opacity="0"
                          width="24"
                        />
                        <path
                          d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                        />
                      </g>
                    </g>
                  </svg>
                </i>
              </span>
            </button>
            <button
              class="item dropdown-item"
              role="menuitem"
              tabindex="0"
              type="button"
            >
              Filters.sortView
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container_inner container"
    >
      <div
        class="no_results"
      >
        <h2>
          Recherche.noResultTitle
        </h2>
        <p>
          Recherche.noResultText
        </p>
        <button
          class="btn login"
        >
          <span
            class="me-2 icon"
          >
            <i
              style="height: 20px; line-height: 20px;"
            >
              <svg
                class="eva eva-refresh-outline"
                fill="#fff"
                height="20px"
                viewBox="0 0 24 24"
                width="20px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-name="Layer 2"
                >
                  <g
                    data-name="refresh"
                  >
                    <rect
                      height="24"
                      opacity="0"
                      width="24"
                    />
                    <path
                      d="M20.3 13.43a1 1 0 0 0-1.25.65A7.14 7.14 0 0 1 12.18 19 7.1 7.1 0 0 1 5 12a7.1 7.1 0 0 1 7.18-7 7.26 7.26 0 0 1 4.65 1.67l-2.17-.36a1 1 0 0 0-1.15.83 1 1 0 0 0 .83 1.15l4.24.7h.17a1 1 0 0 0 .34-.06.33.33 0 0 0 .1-.06.78.78 0 0 0 .2-.11l.09-.11c0-.05.09-.09.13-.15s0-.1.05-.14a1.34 1.34 0 0 0 .07-.18l.75-4a1 1 0 0 0-2-.38l-.27 1.45A9.21 9.21 0 0 0 12.18 3 9.1 9.1 0 0 0 3 12a9.1 9.1 0 0 0 9.18 9A9.12 9.12 0 0 0 21 14.68a1 1 0 0 0-.7-1.25z"
                    />
                  </g>
                </g>
              </svg>
            </i>
          </span>
          Recherche.resetFilters
        </button>
        <div
          class="image"
        >
          <img
            alt="No results"
            data-nimg="1"
            decoding="async"
            height="280"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
            style="color: transparent;"
            width="420"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
